### PR TITLE
Refactors the RSpec test suites and introduces Alma test fixtures

### PR DESCRIPTION
**Pull request by jrgriffiniii**
Advances #1204 and deprecates https://github.com/pulibrary/bibdata/pull/1430

### CODE REVIEW

File: spec/marc_to_solr/lib/config_spec.rb [hunk 1]
```diff
@@ -84,54 +95,61 @@ def fixture_record(fixture_name)
...
     end
   end
 
   describe "electronic_portfolio_s" do
-    it "returns the electronic_portfolio_s field" do
-      record = @indexer.map_record(fixture_record('99122306151806421'))
+    let(:fixture_name) { 'electronic_portfolio' }
+
+    # This test needs to be restored (please see https://github.com/pulibrary/bibdata/issues/1204)
+    xit "returns the electronic_portfolio_s field" do
```
Comment (hackartisan) [inline spec/marc_to_solr/lib/config_spec.rb:152]: Sort of confused -- it looks like the fixture is in this PR -- is the test broken?

File: spec/marc_to_solr/lib/config_spec.rb [hunk 2]
```diff
@@ -169,114 +187,184 @@ def fixture_record(fixture_name)
...
+    let(:fixture_name) { 'scsb_cul_alt_title' }
+
     it 'is a single value for scsb records' do
-      expect(@scsb_alt_title['title_vern_display'].length).to eq(1)
+      expect(record['title_vern_display'].length).to eq(1)
     end
 
-    it 'is a single value for pul records' do
-      expect(@title_vern_display['title_vern_display'].length).to eq(1)
+    context '' do
```
Comment (hackartisan) [inline spec/marc_to_solr/lib/config_spec.rb:333]: Do we know what should go here?

File: spec/marc_to_solr/lib/config_spec.rb [hunk 3]
```diff
@@ -356,172 +449,273 @@ def fixture_record(fixture_name)
...
+      before do
+        ENV['RUN_HATHI_COMPARE'] = 'true'
+      end
+
+      after do
+        ENV['RUN_HATHI_COMPARE'] = ''
+      end
+      let(:fixture_name) { 'hathi_permanent' }
+
+      # This test needs to be restored (please see https://github.com/pulibrary/bibdata/issues/1204)
```
Comment (hackartisan) [inline spec/marc_to_solr/lib/config_spec.rb:521]: Don't worry about this one -- the Hathi functionality won't be used anymore once we launch Alma

**Review by hackartisan — CHANGES_REQUESTED** [review]
The tests read better now and are more descriptive and explicit, thank you.

I left one change request inline.

Otherwise, I don't think I totally understand the newly-pending tests. They seem to have fixtures in this PR -- is it just a matter of reenabling them or is there more to it?

**Review by christinach — CHANGES_REQUESTED** [review]
@jrgriffiniii all the numismatics fixtures should have location_code: `rare$num`. This is the new location for numismatics. I would customize the specific fixtures and update the value of the location_code.

**Comment by jrgriffiniii** [comment]
I have shifted work to a differed (rebased) branch (https://github.com/pulibrary/bibdata/compare/alma...issues-1204-jrgriffiniii-replace-fixtures-fixed), and I am waiting to ensure that the test suites are passing: https://app.circleci.com/pipelines/github/pulibrary/bibdata?branch=issues-1204-jrgriffiniii-replace-fixtures-fixed

**Comment by jrgriffiniii** [comment]
I am sorry, I am deprecating this again in favor of https://github.com/pulibrary/bibdata/pull/1474

**Pull request closed without merging**